client/src/game/ui/HanabiCard.ts

```diff
@@ -196,10 +196,13 @@ export default class HanabiCard extends Konva.Group implements NodeWithTooltip {
   setBareImage() {
     // Optimization: This function is expensive, so don't do it in replays
     // unless we got to the final destination
+    // However, if an action happens before the "turn" message is sent,
+    // we still need to redraw any affected cards
     if (
       this.bareName !== ''
       && globals.replay
-      && globals.replayTurn !== globals.turn
+      && !globals.hypothetical
```

Comment:
oh this is actually unnecessary if we use the inequality, since globals.replayTurn will always be less than or equal to globals.turn in a hypo

Reply:
not necessarily, if we are on the 0th turn of a hypo

Reply:
at the start of a hypo, they are equal, aren't they?

Reply:
im not sure, you could test